packages/HilltopCrawler/src/main/kotlin/nz/govt/eop/hilltop_crawler/fetcher/FetchTasksRunner.kt

```diff
@@ -0,0 +1,50 @@
+package nz.govt.eop.hilltop_crawler.fetcher
+
+import jakarta.annotation.PostConstruct
+import java.time.Duration
+import mu.KotlinLogging
+import org.springframework.scheduling.TaskScheduler
+import org.springframework.stereotype.Component
+
+private const val TASKS_TO_RUN = 10
+
+/**
+ * A class responsible for running fetch tasks using a FetchTaskProcessor.
+ *
+ * Seperated from the processor so that each task processed runs in its own transaction.
+ */
+@Component
+class FetchTasksRunner(val taskScheduler: TaskScheduler, val processor: FetchTaskProcessor) {
+
+  private final val logger = KotlinLogging.logger {}
+
+  /**
+   * Uses springs task scheduler abstraction to effectively start a set of worker threads that will
+   * process tasks from the Hilltop queue.
+   *
+   * When a thread starts working, it will process items from the DB queue until there is nothing
+   * left, when it will stop.
+   */
+  @PostConstruct
+  fun startTasks() {
+    logger.info { "Starting" }
+    (1..TASKS_TO_RUN).forEach { id ->
+      taskScheduler.scheduleWithFixedDelay({ processTasks(id) }, Duration.ofSeconds(10))
+    }
+  }
+
+  fun processTasks(id: Int) {
+    while (true) {
+      try {
+        val hadWorkToDo = processor.runNextTask()
+        if (!hadWorkToDo) {
+          return
+        } else {
+          Thread.yield()
+        }
```

Comment:
What's the different in behaviour here between the `return` and the `Thread.yield()`?

Reply:
TLDR; I'm going to remove the `Thread.yield()` ... me trying to be clever adding it in the first place.

In the return case, which should happen when the queue is empty, it will take 10 seconds before the task is run again to see if there is now work to do.

In the `yield` case, when there is more work to do, its kinda saying "give the other threads a chance to run, but call me back asap" ... and as I've tried to explain that here, I read something that says don't mess with `yield` unless you know you really need it. And because this isn't CPU heavy its probably doing nothing.